js/commerce_bluesnap.form.js

```diff
@@ -0,0 +1,252 @@
+                  $('#bluesnap-cc-type', $form).val(callback.cardData.ccType);
+                  $('#bluesnap-cc-last-4', $form).val(callback.cardData.last4Digits);
+                  $('#bluesnap-cc-expiry', $form).val(callback.cardData.exp);
+
+                  // Finally, submit the form.
+                  $form.get(0).submit();
+                }
+              });
+            }
+          });
+        });
+
+        /**
+         * Define the bluesnapObject.
+         *
+         * Stores Hosted Payment Fields event handlers, styling, placeholder
+         * text, etc.
+         */
+        var bluesnapObject = {
+          onFieldEventHandler: {
+            onFocus: function (tagId) {
+              // Handle focus.
+              changeImpactedElement(tagId, 'hosted-field-valid hosted-field-invalid', 'hosted-field-focus');
+            },
+            onBlur: function (tagId) {
+              // Handle blur.
+              changeImpactedElement(tagId, 'hosted-field-focus');
+            },
+            onError: function (tagId, errorCode, errorDescription) {
+              // Handle a change in validation.
+              changeImpactedElement(tagId, 'hosted-field-valid hosted-field-focus', 'hosted-field-invalid');
+              $('#' + tagId + '-help').removeClass('helper-text-green').text(errorCode + ' - ' + errorDescription);
+
+              // Set the current field complete status to false.
+              bluesnapSetFieldStatus(tagId, false);
+
+              // Display the errors on the form.
+              bluesnapErrorDisplay(bluesnapGetErrorText(errorCode));
+              bluesnapGoToError(tagId);
+            },
+            onType: function (tagId, cardType, cardData) {
+              if (null != cardData) {
+                $('#' + tagId + '-help').addClass('helper-text-green').text(JSON.stringify(cardData));
+              }
+            },
+            onValid: function (tagId) {
+              // Handle a change in validation.
+              changeImpactedElement(tagId, 'hosted-field-focus hosted-field-invalid', 'hosted-field-valid');
+              $('#' + tagId + '-help').text('');
+
+              // Set the current field complete status to true.
+              bluesnapSetFieldStatus(tagId, true);
+
+              // Unset any errors if no elements have the invalid class.
+              if (bluesnapNoErrorsExist()) {
+                $form.find('#payment-errors').html('');
+              }
+            }
+          },
+          // Styling is optional.
+          style: {
+            // Styling all inputs.
+            'input': {
+              'font-size': '14px',
+              'font-family': 'Helvetica Neue,Helvetica,Arial,sans-serif',
+              'line-height': '1.42857143',
+              'color': '#555'
+            },
+            ':focus': {
+              'color': '#555'
+            },
+            '.invalid': {
+              'color': 'red'
+            }
+          },
```

Comment:
Would styling via CSS work? If styles are defined here in JS, would they override any styles defined in CSS? If they won't override CSS and defining styles via CSS for these fields would work then it's ok to define those as defaults. If, however, CSS styles would work without these defaults but they would be overriden by them, the best would be to define no defaults and let the application define the styles via CSS.

Reply:
Styling inside the card fields do not work as specified by BlueSnap:

![image](https://user-images.githubusercontent.com/11439056/56415478-7302be00-62ab-11e9-98f4-e068b0e212ba.png)


I've tried the default styling from BlueSnap, however, it doesn't style/highlight the error fields, and the font is unreadable, etc. So we do need to add the current styles for better UI.